Sources/ITwinMobile/ITMGeoLocationManager.swift

```diff
@@ -391,33 +399,26 @@ public class ITMGeolocationManager: NSObject, CLLocationManagerDelegate, WKScrip
     }
 
     @MainActor
-    private func initAsyncLocationManager() {
-        if _asyncLocationManager == nil {
+    private func requireAsyncLocationManager() -> AsyncLocationManager {
+        if let result = _asyncLocationManager {
+            return result
+        } else {
             _asyncLocationManager = AsyncLocationManager(desiredAccuracy: .bestAccuracy)
+            return _asyncLocationManager!
```

Comment:
I looked for the Swift equivalent of `??=` (nullish coalescing assignment) but couldn't find it.

This is a more compact possibility:
```suggestion
        _asyncLocationManager = _asyncLocationManager ?? AsyncLocationManager(desiredAccuracy: .bestAccuracy)
        return _asyncLocationManager!
```

Reply:
While your suggestion is more compact, I don't like the fact that it always performs an assignment.

Reply:
Yeah I debated that too.

Another alternative:
```suggestion
        if _asyncLocationManager == nil {
            _asyncLocationManager = AsyncLocationManager(desiredAccuracy: .bestAccuracy)
        }
        return _asyncLocationManager!
```